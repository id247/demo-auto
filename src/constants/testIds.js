export default {
  button: 'button',
  select: 'select',
  selectButton: 'select-button',
  input: 'input'
};
