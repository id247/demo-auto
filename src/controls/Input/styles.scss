@import "styles/variables.scss";

.input {
  width: 100%;
  border-radius: $radius-s;
  border: 1px solid $color-grey;
  background: #fff;
  height: 32px;
  padding: 0 $space-s;
  font-size: $font-size-s;
}
