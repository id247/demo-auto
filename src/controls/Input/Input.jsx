import React from 'react';
import testIds from 'constants/testIds';
import './styles.scss';

function Input(props) {
  return (
    <input
      data-testid={testIds.input}
      type="text"
      className="input"
      {...props}
    />
  );
}

export default Input;
