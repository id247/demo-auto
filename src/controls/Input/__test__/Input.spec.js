import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import testIds from 'constants/testIds';
import Input from '../';

afterEach(cleanup);

describe('Input component', () => {
  test('Renders with defaults', async () => {
    const { container, getByTestId } = render(<Input />);

    const input = getByTestId(testIds.input);
    expect(input).toHaveClass('input');
    expect(input).toHaveAttribute('type', 'text');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Handles events', async () => {
    const onChange = jest.fn();
    const event = { target: { value: 'TEST VALUE' } };
    const { getByTestId } = render(<Input onChange={onChange} />);

    const input = getByTestId(testIds.input);
    fireEvent.change(input, event);
    expect(onChange).toHaveBeenCalled();
  });
});
