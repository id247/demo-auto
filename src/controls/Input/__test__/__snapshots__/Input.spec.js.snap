// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Renders with defaults 1`] = `
<input
  class="input"
  data-testid="input"
  type="text"
/>
`;
